packages/utils/Readme.md

```diff
@@ -126,4 +130,136 @@ let count_response: CountResponse = get_count.query(
+
+## Feature Toggle
+
+This module implements feature toggles for your contract. The main motivation behind it is to enable pausing/resuming certain operations rather than stopping/resuming the contract entirely, while providing you with helper functions that will reduce your code to a minimum.
+
+The feature toggles are designed to be flexible, so you can choose whether to put entire messages under a toggle or just certain code sections, etc.
+
+### Initializing Features
+
+Normally you'd want to initialize the features in the `init()` function:
+```rust
+pub fn init<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    msg: InitMsg,
+) -> StdResult<InitResponse> {
+    FeatureToggle::init_features(
+        &mut deps.storage,
+        vec![
+            "Feature1".to_string(),
+            "Feature2".to_string(),
+        ],
+        Some(vec![FeatureStatus::Resumed, // `None` will default to `FeatureStatus::Resumed` for all features
+                  FeatureStatus::Resumed]),
+        vec![env.message.sender], // Can put more than one pauser
+    )?;
+}
+```
+
+### Put a toggle on a message
+
+Putting a toggle on a message (or any code section of your choosing) is as easy as calling `FeatureToggle::require_resumed()`. For example if we have a `Redeem` message in our contract, and we initialized the feature as `"Redeem"`:
+```rust
+fn redeem<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    amount: Option<u128>,
+) -> StdResult<HandleResponse> {
+    FeatureToggle::require_resumed(
+        &deps.storage,
+        vec!["Redeem".to_string()], // you can require more than one toggle here
+    )?;
+    
+    // Continue with function's operation
+}
+```
+If the status of the `"Redeem"` feature is `Stopped`, the contract will error out and stop operation.
+
+### Stop/resume a feature
+
+Firstly, we will need to add `Stop` and `Resume` messages in out `HandleMsg` enum. We can simply use `FeatureToggle::FeatureToggleMsg` - it's an enum that contains default messages that `FeatureToggle` also has default implementation for:
+```rust
+pub enum HandleMsg {
+    // Contract messages
+    Redeem {
+        amount: Option<Uint128>,
+    },
+
+    // Feature toggle
+    Features(FeatureToggleMsg),
+}
+```
+
+The `FeatureToggle` struct contains a default implementation for triggering (stopping/resuming) a feature, so you can just call it from your `handle()` function:
+```rust
+pub fn handle<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    msg: HandleMsg,
+) -> StdResult<HandleResponse> {
+    match msg {
+        HandleMsg::Redeem { amount } => redeem(deps, env, amount),
+        HandleMsg::Features(m) => match m {
+            FeatureToggleMsg::Stop { features } => FeatureToggle::stop(deps, env, features),
+            FeatureToggleMsg::Resume { features } => FeatureToggle::resume(deps, env, features),
+        },
+    }
+}
+```
```

Comment:
`FeatureToggle::{stop, resume}` are `handle_{stop, resume}`. Anyways, developers are forced to use these because the internal implementations `_stop` & `_resume` are not `pub`.

The same issue applies to some other methods that could be `pub`.

Reply:
I intentionally left some functions private, since I thought that if you use this module, you probably don't need it.
Now that the impl. is open for the user to override, it makes more sense (plus it's public by default since the trait is public).